### ISSUE TITLE
MINIO_AUDIT_KAFKA envs not work after RELEASE.2022-02-12T00-51-25Z

### ISSUE THREAD

**Issue by huzhenwei7z** [minio/minio#15764]
We use MINIO_AUDIT_KAFKA environment variables to write the audit log to the Kafka cluster. After we upgraded the minio cluster, we find audit log configurations did not work. After I tested it locally, it works at version `RELEASE.2022-02-07T08-17-33Z` and fails at version `RELEASE.2022-02-12T00-51-25Z`.

## Expected Behavior

```
❯ docker run -p 9000:9000 -p 9001:9001 -e MINIO_AUDIT_KAFKA_ENABLE=on -e MINIO_AUDIT_KAFKA_BROKERS=localhost:1234 -e MINIO_AUDIT_KAFKA_TOPIC=sad quay.io/minio/minio:RELEASE.2022-02-07T08-17-33Z server /data --console-address ":9001"

API: SYSTEM()
Time: 05:13:35 UTC 09/28/2022
DeploymentID: 6e8a1ba3-f712-4d21-9451-ae6f57fcbdca
Error: Unable to initialize server audit Kafka target: dial tcp 127.0.0.1:1234: connect: connection refused (*fmt.wrapError)
       7: cmd/config-current.go:582:cmd.lookupConfigs()
       6: cmd/config-current.go:778:cmd.loadConfig()
       5: cmd/config.go:235:cmd.initConfig()
       4: cmd/config.go:195:cmd.(*ConfigSys).Init()
       3: cmd/server-main.go:417:cmd.initConfigSubsystem()
       2: cmd/server-main.go:352:cmd.initServer()
       1: cmd/server-main.go:549:cmd.serverMain()
WARNING: Detected default credentials 'minioadmin:minioadmin', we recommend that you change these values with 'MINIO_ROOT_USER' and 'MINIO_ROOT_PASSWORD' environment variables
API: http://172.17.0.2:9000  http://127.0.0.1:9000

Console: http://172.17.0.2:9001 http://127.0.0.1:9001

Documentation: https://docs.min.io
```

## Current Behavior

```
❯ docker run -p 9000:9000 -p 9001:9001 -e MINIO_AUDIT_KAFKA_ENABLE=on -e MINIO_AUDIT_KAFKA_BROKERS=localhost:1234 -e MINIO_AUDIT_KAFKA_TOPIC=sad quay.io/minio/minio:RELEASE.2022-02-12T00-51-25Z server /data --console-address ":9001"
API: http://172.17.0.2:9000  http://127.0.0.1:9000

Console: http://172.17.0.2:9001 http://127.0.0.1:9001

Documentation: https://docs.min.io
```

## Possible Solution

make `MINIO_AUDIT_KAFKA` environment variable take effect
## Steps to Reproduce (for bugs)
setup minio as the command above

## Context

We want to enable audit logging via a Kafka topic and have been able to do so. We want to keep the configurations after upgrading instead of finding another way to collect the audit log.

## Your Environment

* Version used (`minio --version`): `RELEASE.2022-02-12T00-51-25Z`
* Server setup and configuration: docker container on a physical machine
* Operating System and version (`uname -a`): Linux U-PF2X4A51 5.4.0-96-generic #109~18.04.1-Ubuntu SMP Thu Jan 13 15:06:26 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

**Comment by vadmeste** [comment]
> docker run -p 9000:9000 -p 9001:9001 -e MINIO_AUDIT_KAFKA_ENABLE=on -e MINIO_AUDIT_KAFKA_BROKERS=localhost:1234 -e MINIO_AUDIT_KAFKA_TOPIC=sad quay.io/minio/minio:RELEASE.2022-02-07T08-17-33Z server /data --console-address ":9001"

Aren't you missing --net=host here ?

**Comment by huzhenwei7z** [comment]
> > docker run -p 9000:9000 -p 9001:9001 -e MINIO_AUDIT_KAFKA_ENABLE=on -e MINIO_AUDIT_KAFKA_BROKERS=localhost:1234 -e MINIO_AUDIT_KAFKA_TOPIC=sad quay.io/minio/minio:RELEASE.2022-02-07T08-17-33Z server /data --console-address ":9001"
> 
> Aren't you missing --net=host here ?

it is just a dummy minio and used only for tests. I didn't deploy Kafka actually, so you don't need to care about it. The key is that the newer version of minio doesn't report similar errors as the old version. I suspect the environment variables are ignored.

**Comment by harshavardhana** [comment]
fixed via #15766
